spec/article_json/import/google_doc/text_element_spec.rb

```diff
@@ -0,0 +1,163 @@
+    end
+
+    context 'when the node has a bold & italic class attribute' do
+      let(:xml_fragment) { '<span class="bold-italic">foo bar</span>' }
+      it { should be true }
+    end
+
+    context 'when the node has a non-italic class attribute' do
+      let(:xml_fragment) { '<span class="bold">foo bar</span>' }
+      it { should be false }
+    end
+
+    context 'when the node has no class attribute' do
+      let(:xml_fragment) { '<span>foo bar</span>' }
+      it { should be false }
+    end
+  end
+
+  describe '#href' do
+    subject { element.href }
+
+    context 'when the text includes no link' do
+      let(:xml_fragment) { '<span>foo bar</span>' }
+      it { should be nil }
+    end
+
+    context 'when the text includes a simple link' do
+      let(:xml_fragment) do
+        <<-html
+          <span><a href="https://devex.com">foo bar</a></span>
+        html
+      end
+      it { should eq 'https://devex.com' }
+    end
+
+    context 'when the text includes a link with a google redirect' do
+      let(:xml_fragment) do
+        <<-html
+          <span><a href="https://www.google.com/url?q=https://devex.com">foo bar</a></span>
+        html
+      end
+      it { should eq 'https://devex.com' }
+    end
+  end
+
+  describe 'to_h' do
+    subject { element.to_h }
+
+    context 'when having a simple text' do
+      let(:xml_fragment) { '<span>foo bar</span>' }
+      it 'returns a proper Hash' do
+        expect(subject).to be_a Hash
+        expect(subject[:type]).to eq :text
+        expect(subject[:content]).to eq 'foo bar'
+        expect(subject[:bold]).to be false
+        expect(subject[:italic]).to be false
+        expect(subject[:href]).to be nil
+      end
+    end
+
+    context 'when having a styles text and a link' do
+      let(:xml_fragment) do
+        <<-html
+          <span class="bold-italic"><a href="https://devex.com">foo bar</a></span>
+        html
+      end
+      it 'returns a proper Hash' do
+        expect(subject).to be_a Hash
+        expect(subject[:type]).to eq :text
+        expect(subject[:content]).to eq 'foo bar'
+        expect(subject[:bold]).to be true
+        expect(subject[:italic]).to be true
+        expect(subject[:href]).to eq 'https://devex.com'
+      end
+    end
+  end
+
+  describe '.extract' do
+    subject do
+      described_class.extract(text_node: nokogiri_node,
+                              css_analyzer: css_analyzer)
+    end
+
+    let(:xml_fragment) do
+      <<-html
+        <p><span class="bold">A </span><span><a href="https://devex.com">link</a></span><span class="italic"> and styling.</span></p>
+      html
+    end
+
+    it 'parses all text sub-nodes' do
+      puts subject.map(&:to_h)
+      expect(subject.size).to eq 3
+
+      expect(subject[0].content).to eq 'A '
+      expect(subject[0].bold?).to be true
+      expect(subject[0].italic?).to be false
+      expect(subject[0].href).to be nil
+
+      expect(subject[1].content).to eq 'link'
```

Comment:
Since these are all small unit tests, I think you could have separate `it` cases for the different expectancies.

Reply:
I thought about it but then decided to have them properly grouped in an `it` block... No big deal IMO.

Reply:
Also, I wouldn't know how to do that without using `its`, which is no longer a part of rspec-core => 3.x:  https://gist.github.com/myronmarston/4503509

Reply:
Well, you would have to do it with regular `it` blocks.. I always like more atomic tests, but I know that not everyone agrees with that here :)